src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -2503,6 +2505,30 @@ void ReadParameterStatus(ReadOnlySpan<byte> incomingName, ReadOnlySpan<byte> inc
             case "TimeZone":
                 Timezone = value;
                 return;
+
+            case "default_transaction_read_only":
+                _transactionReadOnly = value == "on";
+                UpdateClusterState(this);
+                return;
+
+            case "in_hot_standby":
+                _hotStandby = value == "on";
+                UpdateClusterState(this);
+                return;
+            }
+
+            static void UpdateClusterState(NpgsqlConnector connector)
```

Comment:
Seems like this could be extracted out and used from QueryClusterState as well, to avoid duplication